### DISCUSSION TITLE
4 Shadow modes

### DISCUSSION THREAD

**Discussion by imakihi** [TerriaJS/terriajs#5649] [Q&A] [answered]
Hi there. I am testing the 4 shadow mode with my building data (see my attachment). I am not sure the "Receive Only" mode is working correctly. As far as I can see, the Receive Only mode resulted in exactly the same as "None". Does anybody has experienced the same issue? Or do I misunderstand the difference between "None" and "Receive Only" modes?

<img width="1152" alt="shadowmode" src="https://user-images.githubusercontent.com/4824436/126199752-30dee000-a886-4cc4-94c1-283098aade71.png">

Thank you!

**Comment by na9da** [comment, marked as answer]
Hi @imakihi,

As described in the Cesium doc, [RECEIVE_ONLY](https://cesium.com/learn/cesiumjs/ref-doc/global.html?classFilter=shadowm#ShadowMode) means the model can receive shadows from other models but does not cast any of its own. For the screenshot with `RECEIVE_ONLY`, the tileset as a whole can receive shadows from other tilesets or models, but wont cast any shadows of its own. Because there are no nearby models or tilesets to cast a shadow, it looks just like the `NONE` example.

Below is an example screenshot of the `RECEIVE_ONLY` mode in action. The tileset has `RECEIVE_ONLY`  and the GLTF model is casting a shadow on it.

![image](https://user-images.githubusercontent.com/1430/126240842-cbc3a448-6707-43d3-b9b6-1cfcdbbeaf3c.png)


Contrast it with this one where the tileset shadow mode is `NONE` so it doesn't "receive" the shadow of the adjacent gltf model.

![image](https://user-images.githubusercontent.com/1430/126241490-63ffa9f3-9178-4001-9bf1-c3925c91eacc.png)


Hope that answers your question.

**Reply by imakihi** [reply]
@na9da Thank you so much for getting back on this.
That is interesting. So, I experimented a bit more with my 3D Tiles. In this video, you two building layers. A border of these two layers are the road in the middle of the map.

https://www.screencast.com/t/1GlFQc659

In this video, I don't see any differences between None and Receive Only. What do you think?

Hiroo

**Reply by meh9** [reply]
In the video there is one combination left that you didn't try, and that is to have one of the datasets to have one of the Cast options, and the other to have Receive - in this configuration the Cast dataset will project shadows on the Receive dataset. :)

**Reply by imakihi** [reply]
Hi @meh9 . Thank you for your suggestion! I tested the combination and realized that combination works. Just in case, here is my test result.
https://www.screencast.com/t/Zi3LJnFZc

@na9da @meh9 Thank you so much!!

Best.

Hiroo
